test/integration/Swap.spec.ts

```diff
@@ -0,0 +1,120 @@
+import LndClient, { LndClientConfig } from '../../lib/lndclient/LndClient';
+import Logger, { Level } from '../../lib/Logger';
+import DB from '../../lib/db/DB';
+import * as lndrpc from '../../lib/proto/lndrpc_pb';
+import { waitForSpy } from '../utils';
+
+describe('Swaps.Integration', () => {
+  const loggers = Logger.createLoggers(Level.Warn);
+  let db: DB;
+  let pool: Pool;
+  let swaps: Swaps;
+  let lndBtcClient: LndClient;
+  let lndLtcClient: LndClient;
+  let peer: Peer;
+
+  before(async () => {
+    db = new DB(loggers.db);
+    await db.init();
+    // peer
+    peer = sinon.createStubInstance(Peer) as any;
+    peer.sendPacket = () => {};
+    peer.getLndPubKey = () => '1234567890';
+    // pool
+    pool = sinon.createStubInstance(Pool) as any;
+    pool.getPeer = () => peer;
+    // queryRoutes responses
+    const queryRoutesSuccessResponse = () => {
+      return Promise.resolve({
+        getRoutesList: () => [1],
+      } as any);
+    };
+    // lnd btc
+    lndBtcClient = sinon.createStubInstance(LndClient) as any;
+    lndBtcClient.queryRoutes = queryRoutesSuccessResponse;
+    lndBtcClient.isConnected = () => true;
+    // lnd ltc
+    lndLtcClient = sinon.createStubInstance(LndClient) as any;
+    lndLtcClient.isConnected = () => true;
+    lndLtcClient.queryRoutes = queryRoutesSuccessResponse;
+    swaps = new Swaps(loggers.swaps, db.models, pool, lndBtcClient, lndLtcClient);
+  });
+
+  after(async () => {
+    await db.close();
+  });
+
+  describe('executeSwap', () => {
+
+    it('will resolve valid maker and taker order', async () => {
+      const maker = {
+        price: 0.008,
+        quantity: 0.01,
+        initialQuantity: 0.01,
+        pairId: 'LTC/BTC',
+        isBuy: false,
+        peerPubKey: '029706a877a39d9f22e1ecd1042b058bd5798b9fd79fcf8c8ced8c1f13a4aff055',
+        id: 'b76158b1-e34c-11e8-92c2-9fa16c9aa9d1',
+        createdAt: 1541677927103,
+      };
+      const taker = {
+        pairId: 'LTC/BTC',
+        price: 0.008,
+        quantity: 0.01,
+        initialQuantity: 0.01,
+        hold: 0,
+        isBuy: true,
+        localId: 'bd1d6500-e34c-11e8-bedb-9ffbf8c8cc23',
+        id: 'bd1d6501-e34c-11e8-bedb-9ffbf8c8cc23',
+        createdAt: 1541677936464,
+      };
+      const swapResult = {
+        orderId: '760d5291-e43e-11e8-bd56-e5c08173fa7d',
+        localId: '76c61b40-e43e-11e8-a3b5-853f31e7d8e6',
+        pairId: 'LTC/BTC',
+        quantity: 0.00001,
+        amountReceived: 1000,
+        amountSent: 8,
+        rHash: 'd94c22a73d2741ed5cdcf3714f9ab3c8664793b03a54c74a08877726007d67c2',
+        peerPubKey: '020c9a0fb8dac5b91756fb21509aefc4e95b585510c4de6e6311f18348a4723cdd',
+        role: 0,
+      };
+      const swapListenersAdded = sinon.spy(swaps, 'on');
+      const addDealSpy = sinon.spy(swaps, 'addDeal');
+      const swapListenersRemoved = sinon.spy(swaps, 'removeListener');
+      expect(swaps.executeSwap(maker, taker)).to.eventually.equal(swapResult);
```

Comment:
I believe you should be `await`ing this since the assertion is a promise in this case.

Reply:
I can't use `await swaps.executeSwap(maker, taker);` here because it would block the execution of the test function until `executeSwap` resolves. It will never happen because `swap.paid` event is not emitted.

`chai-as-promised` allows us to continue executing the test, but still assert `to.eventually.equal` that the promise eventually resolves.

Reply:
Ok sounds good, thanks for the explanation.